scripts/vm_run_tests.psm1

```diff
@@ -31,44 +31,244 @@ function Invoke-CICDTestsOnVM
-            Write-Log "Invoking xdp reflect test (IPv4)."
-            .\xdp_tests.exe xdp_reflect_test --remote-ip $RemoteIPV4Address | Write-Log
-            Write-Log "Invoking xdp reflect test (IPv6)."
-            .\xdp_tests.exe xdp_reflect_test --remote-ip $RemoteIPV6Address | Write-Log
-            Write-Log "xdp reflect tests passed." -ForegroundColor Green
-            popd
-    } -ArgumentList ("C:\eBPF", ("{0}_{1}" -f $VM1, $LogFileName), $VM1V4Address, $VM1V6Address) -ErrorAction Stop
+    } -ArgumentList ($VM, "C:\eBPF", ("{0}_{1}" -f $VM, $LogFileName)) -ErrorAction Stop
+}
+
+function Invoke-XDPTest1
+{
+    param([Parameter(Mandatory=$True)] [string] $VM1,
+          [Parameter(Mandatory=$True)] [string] $VM2,
+          [Parameter(Mandatory=$True)] [string] $VM1Int1V4Address,
+          [Parameter(Mandatory=$True)] [string] $VM1Int1V6Address,
+          [Parameter(Mandatory=$True)] [string] $VM1Int2V4Address,
+          [Parameter(Mandatory=$True)] [string] $VM1Int2V6Address,
+          [Parameter(Mandatory=$True)] [string] $LogFileName)
+
+    # Load reflect_packet without specifying interface on VM1.
+    $ProgId = Add-eBPFProgrmOnVM -VM $VM1 -Program "reflect_packet.o" -LogFileName $LogFileName
+
+    # Run XDP reflect test from VM2 targeting both interfaces of VM1.
+    Invoke-XDPTestOnVM $VM2 "xdp_reflect_test" $VM1Int1V4Address $VM1Int1V6Address $LogFileName
+    Invoke-XDPTestOnVM $VM2 "xdp_reflect_test" $VM1Int2V4Address $VM1Int2V6Address $LogFileName
+
+    # Unload program from VM1.
+    Remove-eBPFProgramFromVM $VM1 $ProgId $LogFileName
+}
+
+function Invoke-XDPTest2
+{
+    param([Parameter(Mandatory=$True)] [string] $VM1,
+          [Parameter(Mandatory=$True)] [string] $VM2,
+          [Parameter(Mandatory=$True)] [string] $VM1Int1Alias,
+          [Parameter(Mandatory=$True)] [string] $VM1Int2Alias,
+          [Parameter(Mandatory=$True)] [string] $VM1Int1V4Address,
+          [Parameter(Mandatory=$True)] [string] $VM1Int1V6Address,
+          [Parameter(Mandatory=$True)] [string] $VM1Int2V4Address,
+          [Parameter(Mandatory=$True)] [string] $VM1Int2V6Address,
+          [Parameter(Mandatory=$True)] [string] $LogFileName)
+
+    # Load reflect_packet on interface1 on VM1.
+    $ProgId = Add-eBPFProgrmOnVM -VM $VM1 -Program "reflect_packet.o" -Interface $VM1Int1Alias -LogFileName $LogFileName
+
+    # Attach the program on interface2 on VM1.
+    Set-eBPFProgrmOnVM -VM $VM1 -ProgId $ProgId -Interface $VM1Int2Alias -LogFileName $LogFileName
+
+    # Run XDP reflect test from VM2 targeting both interfaces of VM1.
+    Invoke-XDPTestOnVM $VM2 "xdp_reflect_test" $VM1Int1V4Address $VM1Int1V6Address $LogFileName
+    Invoke-XDPTestOnVM $VM2 "xdp_reflect_test" $VM1Int2V4Address $VM1Int2V6Address $LogFileName
+
+    # Unload program from VM1.
+    Remove-eBPFProgramFromVM $VM1 $ProgId $LogFileName
+}
+
+function Invoke-XDPTest3
+{
+    param([Parameter(Mandatory=$True)] [string] $VM1,
```

Comment:
style question: the ebpf style guide (mainly for C code) normally says to start variables with lower case.  Does powershell require a different convention?   It's good to keep consistency with the rest of this file, but it looks like this file doesn't follow the currently stated ebpf coding conventions so want to know if they wouldn't apply to powershell scripts.

Reply:
I thought snake_case was clashing with the casing style of PS cmdlets. So I chose this casing. I can update the deploy_ebpf.ps1 to use different casing. But yes, I would like styling guidance of PS scripts to not use snake_case.